design/plugininputs-field.md

```diff
@@ -0,0 +1,115 @@
+
+## Goals
+- Add pluginInputs field in the velero CRs (Backup, Schedule, Restore) to allow the user to specify the parameters which are specific to a plugin.
+
+# Precedence of this Design principle 
+This approach is similar to all current velero / K8s storage design principles.
+- **BackupStorageLocation** - we refer the name of the object store plugin provider(for eg "azure"), and below under config, we can specify the config specific to the plugin which is a generic property bag. Only the specific object store plugin can understand the config.
+- **CSI based VolumeSnapshotClass** - we refer the name of the CSI driver(for eg "csi.cloud.disk.driver"), and below under parameters, we can specify the parameters specific to the plugin which is a generic property bag. Parameters such as SKU etc are only understood by the specific CSI driver.
+- **VolumeSnapshotLocation** - we refer the name of the volume snapshotter plugin provider(for eg "aws"), and below under config, we can specify the config specific to the plugin which is a generic property bag. Properties such as region etc are only understood by the specific volume snapshotter plugin.
+
+
+### Use case examples for other plugins
+
+#### Use case 1: Change StorageClass RIA (velero.io/change-storage-class)
+Change Storage Class plugin, uses a global configmap which is discovered through labels/ annotations which is not ideal customer experience and prohibhit usage of specific settings for each backup/ schedule setup.
+
+The pluginInputs field can be used to pass in the reference the configmap/ secret to use for a particular plugin. This will allow the user to use different configmaps/ secrets for different backups/ schedules.
+
+Example: 
+```yaml
+spec:
+    pluginInputs:
+    - name: velero.io/storageclass
+    - properties:
+        - key: storageclass-mapping-configmap
+        - value: configMapNamespace/configMapName
+```
+
+#### Use case 2: Change Image Name (velero.io/change-image-name)
+This plugin is similar to velero.io/change-storage-class plugin and uses a global configmap. 
+
+Even the code uses below logic and fails if multiple configmaps.
+```
+return nil, errors.Errorf("found more than one ConfigMap matching label selector %q: %v", opts.LabelSelector, items)
+```
+
+Providing reference of configmap in the pluginInputs field will allow the user to use different configmaps for different restores.
+
+#### Use case 3: Support for multiple CSI VolumeSnapshotClass (velero.io/csi)
+Refer PR: https://github.com/vmware-tanzu/velero/pull/5774
+Refer Issue: https://github.com/vmware-tanzu/velero/issues/5750
+
+Currently the Velero CSI plugin chooses the VolumeSnapshotClass in the cluster that has the same driver name and also has the velero.io/csi-volumesnapshot-class label set on it. This global selection is not sufficient for many use cases.
+
+Example: 
+```yaml
+apiVersion: velero.io/v1
+kind: Backup
+metadata:
+  name: backup-1
+spec:
+    pluginInputs:
+    - name: velero.io/csi
```

Comment:
The plugin name should be `velero.io/csi-volumesnapshot-backupper`.
I also wonder if the configuration is applied to many plugins, and we can only choose the plugin by name, then this gonna be a long list.
Could we consider adding some RE function here?